src/core/functions/internal_functions/system/MultiSelectModal.ts

```diff
@@ -0,0 +1,229 @@
+import { Modal, TextComponent, fuzzySearch, prepareQuery } from "obsidian";
+import { TemplaterError } from "utils/Error";
+
+export class MultiSelectModal<T> extends Modal {
+    private resolve: (values: T[]) => void;
+    private reject: (reason?: TemplaterError) => void;
+    private selectedItems = new Set<string>();
+    private mappedItems = new Map<string, T>();
+    private submitted = false;
+
+    private SELECTOR_SELECTED = ".is-selected";
+    private CLASS_SELECTED = "is-selected";
+
+    private suggestionsContainer: HTMLDivElement;
+    private chipsContainer: HTMLDivElement;
+    private textInput: TextComponent;
+
+    constructor(
```

Comment:
Ideally, the constructor should be super simple and the content creation should be delegated to a separate function. The reasoning behind this is that separate functions are easier to test and debug and a constructor should only really take care of initializing of the core components, in this case perhaps just call `super()` like the other modals in the project. Would also be nice for consistency.

Reply:
Sure, I will put all initialization code from constructor to a separate function